packages/amplify-cli-core/src/overrides-manager/override-runner.ts

```diff
@@ -0,0 +1,11 @@
+import { $TSAny, $TSContext, AmplifyError, pathManager } from '../index';
+import * as path from 'path';
+
+export const runOverride = (overrideDir: string, templateObject: $TSAny, projectInfo: $TSAny): void => {
+  const overrideJSFilePath = path.join(overrideDir, 'build', 'override.js');
+  delete require.cache[require.resolve(overrideJSFilePath)];
+  const overrideImport = require(overrideJSFilePath);
+  if (overrideImport && overrideImport?.override && typeof overrideImport?.override === 'function') {
+    overrideImport.override(templateObject, projectInfo);
+  }
+};
```

Comment:
In some places we had.
```
await sandboxNode
            .run(overrideCode, overrideJSFilePath)
            .override(this._resourceTemplateObj as AmplifyDDBResourceTemplate, projectInfo);
```

we should make this function async and `await overrideImport.override(templateObject, projectInfo);`.

This works even if `overrideImport.override` is sync and returns `void` . The way compiler handles `awaits` is that it inspects return object coming from function at runtime.